### DISCUSSION TITLE
ADIF export: single line per QSO?

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#3099] [Ideas / Feature Requests]
Is it possible to redesign the ADIF export to produce one line per QSO?
Now it produces one line per tag (parameter).

**Comment by magicbug** [comment]
Anything is possible its just line formatting however I'm interested in the reason behind wanting it

**Reply by R1BLH** [reply]
1. Some adif parsers handle the linefeed character (0x0A) in the QSO string incorrectly.
2. it is easier, at least for me, to visually view the file if the record of a QSO is in one line.

**Comment by phl0** [comment]
> 1. Some adif parsers handle the linefeed character (0x0A) in the QSO string incorrectly.

These should just be ignored. A valid ADIF records end in <eor> regardless of newlines and/or linefeeds.

**Reply by R1BLH** [reply]
I agree. If I had written the parser myself, there would be no questions. But I didn't write it. :)
In particular, I got this problem when converting to .edi format (VHF contest report).

**Comment by magicbug** [comment]
If you do want to edit the code then i'd start here https://github.com/magicbug/Cloudlog/blob/master/application/libraries/AdifHelper.php

**Reply by R1BLH** [reply]
Thank you! I'll try!

**Reply by R1BLH** [reply]
@magicbug I did it! But should it be merged into the development branch? Is it suitable for everyone?
I tried to use a variable from config.php to control the output, but was unsuccessful. 
Could you advice how to get a variable from the config.php file into the AdifHelper.php?
